// docusign-rooms/src/model/FormLibraries.js
/*
 * DocuSign Rooms API - v2
 * An API for an integrator to access the features of DocuSign Rooms
 *
 * OpenAPI spec version: v2
 * Contact: devcenter@docusign.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.13-SNAPSHOT
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Docusign) {
      root.Docusign = {};
    }
    root.Docusign.FormLibraries = factory(root.Docusign.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The FormLibraries model module.
   * @module model/FormLibraries
   * @version 3.0.0
   */

  /**
   * Constructs a new <code>FormLibraries</code>.
   * The `FormLibraries` resource enables you to access standard real estate industry association forms and add them to rooms. 
   * @alias module:model/FormLibraries
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>FormLibraries</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/FormLibraries} obj Optional instance to populate.
   * @return {module:model/FormLibraries} The populated <code>FormLibraries</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('formsLibraryId'))
        obj.formsLibraryId = ApiClient.convertToType(data['formsLibraryId'], 'String');
      if (data.hasOwnProperty('name'))
        obj.name = ApiClient.convertToType(data['name'], 'String');
      if (data.hasOwnProperty('formCount'))
        obj.formCount = ApiClient.convertToType(data['formCount'], 'Number');
    }
    return obj;
  }

  /**
   * The id of the form library.  Example: `402c6e2f-xxxx-xxxx-xxxx-ff3f249f6da9` 
   * @member {String} formsLibraryId
   */
  exports.prototype.formsLibraryId = undefined;

  /**
   * The name of the office.
   * @member {String} name
   */
  exports.prototype.name = undefined;

  /**
   * The number of forms in the form library.   Example: `50`
   * @member {Number} formCount
   */
  exports.prototype.formCount = undefined;

  return exports;

}));
