// docusign-rooms/src/model/RoomTemplates.js
/*
 * DocuSign Rooms API - v2
 * An API for an integrator to access the features of DocuSign Rooms
 *
 * OpenAPI spec version: v2
 * Contact: devcenter@docusign.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.13-SNAPSHOT
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Docusign) {
      root.Docusign = {};
    }
    root.Docusign.RoomTemplates = factory(root.Docusign.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The RoomTemplates model module.
   * @module model/RoomTemplates
   * @version 3.0.0
   */

  /**
   * Constructs a new <code>RoomTemplates</code>.
   * The room template resources provides a method that enables you to retrieve the room templates associated with an account.
   * @alias module:model/RoomTemplates
   * @class
   */
  var exports = function() {
  };

  /**
   * Constructs a <code>RoomTemplates</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/RoomTemplates} obj Optional instance to populate.
   * @return {module:model/RoomTemplates} The populated <code>RoomTemplates</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();
      if (data.hasOwnProperty('roomTemplateId'))
        obj.roomTemplateId = ApiClient.convertToType(data['roomTemplateId'], 'Number');
      if (data.hasOwnProperty('name'))
        obj.name = ApiClient.convertToType(data['name'], 'String');
      if (data.hasOwnProperty('taskTemplateCount'))
        obj.taskTemplateCount = ApiClient.convertToType(data['taskTemplateCount'], 'Number');
    }
    return obj;
  }

  /**
   * The id of the room template.
   * @member {Number} roomTemplateId
   */
  exports.prototype.roomTemplateId = undefined;

  /**
   * The name of the room template.
   * @member {String} name
   */
  exports.prototype.name = undefined;

  /**
   * The total number of task templates that the room template uses.
   * @member {Number} taskTemplateCount
   */
  exports.prototype.taskTemplateCount = undefined;

  return exports;

}));
